python/taichi/graph/_graph.py

```diff
@@ -97,25 +99,39 @@ def run(self, args):
 def Arg(tag,
         name,
         dtype=None,
-        field_dim=0,
+        ndim=0,
+        field_dim=None,
         element_shape=(),
         channel_format=None,
         shape=(),
         num_channels=None):
+    if field_dim is not None:
+        if ndim != 0:
+            raise TaichiRuntimeError(
+                'field_dim is deprecated, please do not specify field_dim and ndim at the same time.'
+            )
+        warnings.warn(
+            "The field_dim argument for ndarray will be deprecated in v1.5.0, use ndim instead.",
```

Comment:
Should we add an test to `test_deprecation.py`?

Reply:
Thanks, added the deprecation tests!